api/user.go

```diff
@@ -340,5 +347,33 @@ func (a *UserAPI) UpdateCurrentUser(c *gin.Context) {
 		}
 	}
 
+	if updateUserRequest.GitLabToken != nil {
```

Comment:
This block and the `GitHubToken` could be merged into a function with the provider proper parameters.